### PR TITLE
extend ftp.py to pass 'callback' to ftplib.storbinary

### PR DESCRIPTION
**Pull request by ipmonk**
This is to facilitate ftp progress reporting

### CODE REVIEW

**Comment by jnpr-community-netdev** [comment]
Can one of the admins verify this patch?

**Comment by vnitinv** [comment]
ok to test

File: lib/jnpr/junos/utils/ftp.py [hunk 1]
```diff
@@ -68,7 +68,12 @@ def put(self, local_file, remote_path=None):
                     remote_file = os.path.join('/tmp/', local_file)
                 else:
                     remote_file = os.path.join(remote_path, local_file)
-            self.storbinary('STOR ' + remote_file, open(local_file, 'rb'))
+
+            if 'callback' in self._ftpargs:
+                self.storbinary('STOR ' + remote_file, open(local_file, 'rb'),
```
Comment (vnitinv) [inline lib/jnpr/junos/utils/ftp.py]: Hi @ipmonk

Can you change the call to 
self.storbinary(cmd='STOR ' + remote_file, fp=open(local_file, 'rb'), callback=self._ftpargs['callback'])

We don't need to pass blocksize from here.

**Comment by vnitinv** [comment]
If user is not passing `callback` parameter, the function call will error out. You can change it to

callback=self._ftpargs.get('callback')

**Comment by jnpr-community-netdev** [comment]
Can one of the admins verify this patch?

**Pull request merged**